src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -415,4 +418,20 @@ default boolean damage(double damage, DamageSource damageSource) {
         return getWorld().getEntities(predicate::test);
     }
 
+    /**
+     * Gets the current display name for this {@link Entity}.
+     *
+     * @return The display name
+     */
+    Text getDisplayName();
```

Comment:
> All entities have a display name, and vanilla itself will break if it is not present/is null.

That's a completely different name then. Something that would make sense in the `TextRepresentable` bit. There was a PR somewhere adding that functionality (if it wasn't already added). The issue with this method being named "displayName" is that it's clashing with the idea of the `Keys.DISPLAY_NAME` since both use different fields/texts/formatting.